hudi-client/hudi-java-client/src/main/java/org/apache/hudi/client/HoodieJavaWriteClient.java

```diff
@@ -140,6 +140,19 @@ public boolean commit(String instantTime,
     return postWrite(result, instantTime, table);
   }
 
+  @Override
+  public List<WriteStatus> insertOverwrite(List<HoodieRecord<T>> records, String instantTime) {
+    HoodieTable<T, List<HoodieRecord<T>>, List<HoodieKey>, List<WriteStatus>> table =
+        getTableAndInitCtx(WriteOperationType.INSERT_OVERWRITE, instantTime);
+    table.validateUpsertSchema();
```

Comment:
We should validate insert schema here right?

Reply:
I got confused with the `insert_overwrite` , `insert_overwrite_table` and `upsert` operation!
so I used `validateUpsertSchema()` .
I will make change